### PR TITLE
fix(simulation): store uploaded shards the way CAS does for LocalClient

### PR DESCRIPTION
**Pull request by sirahd**
The simulation rebuilt each uploaded shard through `MDBInMemoryShard`, while CAS re-serializes through the streaming serializer and keys the object by the hash of those bytes. Only the streaming path stamps `shard_creation_timestamp`, so identical content collapsed onto a single hash forever in the simulation but not in production. That let a re-upload land back on a hash GC had already deregistered from the global dedup table and still had queued for deletion, leaving dedup rows pointing at a condemned shard — `stress_test_high_fp` failed on it 7 runs out of 7.

Two consequences of storing what CAS stores:
- shards carry no lookup tables, so `get_file_chunk_hashes` reads its file entry from the authoritative table, as the reconstruction path already did
- they record a zero key expiry, so this server-side store is no longer scanned on open as if it were a client cache — that scan prunes by exactly that field

Test plan: full GC stress suite against this branch with no GC-side change, all three variants green and `integrity=36ok/0FAIL` each (`stress_test_high_fp` 14/14 epochs, `stress_test`, `stress_test_convergence`), at unchanged throughput and unchanged collection rates.

---

> [!NOTE]
> **Medium Risk**
> Changes simulation shard hashing, on-disk layout, and file lookup paths that GC stress tests depend on; behavior is intentional parity with CAS but affects dedup/GC edge cases in tests only.
> 
> **Overview**
> Aligns **simulation `LocalClient` shard persistence** with production CAS so GC stress tests match real dedup/GC behavior.
> 
> **Shard uploads** no longer rebuild via `MDBInMemoryShard`; validated input is re-serialized through the streaming path (`HashedWrite` + `minimal_shard.serialize`), keyed by the hash of those bytes (client-supplied hash ignored), written atomically via temp file + rename. Identical logical content therefore gets a **new shard hash** when the footer stamps a creation timestamp—fixing cases where re-upload reused a hash already deregistered/queued for deletion while dedup rows still pointed at it.
> 
> **Shard manager init** disables directory scan on open (`scan: false`) because CAS-form shards use zero key expiry; scanning would incorrectly prune them as if this were an expiring client cache.
> 
> **`get_file_chunk_hashes`** resolves file metadata from **`FILE_TO_SHARD_TABLE`** (`get_file_info_from_table`) instead of shard-manager reconstruction lookup, consistent with shards stored without embedded lookup tables.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5cec0e0586d2efafae5dfa588abe6886188e7404. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: xet_client/src/cas_client/simulation/local_client.rs [hunk 1]
```diff
@@ -1522,34 +1526,33 @@ impl Client for LocalClient {
...
-            in_memory_shard.add_file_reconstruction_info(MDBFileInfo::from(file_view))?;
-        }
+        // Store the shard exactly as CAS does: re-serialize the validated shard through the
+        // streaming serializer and store those bytes under their own hash, discarding the
+        // hash the client sent (see `validate_shard_from_async_read` in cas_server). That
+        // footer stamps a creation timestamp, so re-uploading identical content produces a
+        // distinct object rather than reusing one hash forever.
+        let mut cas_form = Vec::with_capacity(minimal_shard.serialized_size(false));
+        let mut hashed_writer = HashedWrite::new(&mut cas_form);
+        minimal_shard.serialize(&mut hashed_writer, false)?;
```
[the remark this answers is not in the document]
Comment (sirahd) [reply xet_client/src/cas_client/simulation/local_client.rs]: Good catch — confirmed and fixed in f62f815f.

The chain is real: `From<&MDBFileInfoView> for MDBFileInfo` preserves verification and `MDBFileInfo::serialize` writes it, so the previous in-memory path did store those entries, and re-serializing with verification off dropped them.

Worth noting why matching CAS exactly is not the fix here. CAS trims verification too, but it serves file info from its own database (`cas_db.get_file_reconstruction`), which retains it — in the simulation the stored shard *is* what file lookups read, so trimming loses data production never loses. So verification is now kept on write.

One wrinkle: `serialize` returns `missing requested verification info` unless every file carries verification, so the flag is computed from the shard rather than hardcoded. A shard mixing files with and without verification still stores none, since the API has no per-file control — same as before this PR.

Added `test_upload_shard_keeps_file_verification`, which fails without the fix with `lost its verification entries on upload`.

File: xet_client/src/cas_client/simulation/local_client.rs [hunk 2]
```diff
@@ -1522,34 +1526,40 @@ impl Client for LocalClient {
...
-
-        // Add file info from the views
-        for i in 0..minimal_shard.num_files() {
-            let file_view = minimal_shard.file(i).unwrap();
-            in_memory_shard.add_file_reconstruction_info(MDBFileInfo::from(file_view))?;
-        }
+        // Store the shard exactly as CAS does: re-serialize the validated shard through the
+        // streaming serializer and store those bytes under their own hash, discarding the
+        // hash the client sent (see `validate_shard_from_async_read` in cas_server). That
+        // footer stamps a creation timestamp, so re-uploading identical content produces a
```
Comment (seanses) [inline xet_client/src/cas_client/simulation/local_client.rs:1532]: What's the intent of this PR? I thought we actually don't want this behavior, i.e. re-uploading identical content produces a distinct object. And we can fix it by enforcing the timestamp in the serialize shard footer to be, for example, all zeros.

**Review by seanses — APPROVED** [review]

File: xet_client/src/cas_client/simulation/local_client.rs (hunk 2 above)
Comment (seanses) [reply xet_client/src/cas_client/simulation/local_client.rs:1532]: Communicated offline.
